hackernews_tui/src/client/mod.rs

```diff
@@ -376,27 +376,65 @@ impl HNClient {
         let output = std::process::Command::new(&article_parse_command.command)
             .args(&article_parse_command.options)
             .arg(url)
-            .output()?;
-
-        if output.status.success() {
-            match serde_json::from_slice::<Article>(&output.stdout) {
-                Ok(mut article) => {
-                    // Replace a tab character by 4 spaces as it's possible
-                    // that the terminal cannot render the tab character.
-                    article.content = article.content.replace('\t', "    ");
-
-                    article.url = url.to_string();
-                    Ok(article)
-                }
-                Err(err) => {
-                    let stdout = std::str::from_utf8(&output.stdout)?;
-                    warn!("failed to deserialize {} into an `Article` struct:", stdout);
-                    Err(anyhow::anyhow!(err))
+            .output();
+
+        match output {
+            Ok(output) => {
+                if output.status.success() {
+                    match serde_json::from_slice::<Article>(&output.stdout) {
+                        Ok(mut article) => {
+                            // Replace a tab character by 4 spaces as it's possible
+                            // that the terminal cannot render the tab character.
+                            article.content = article.content.replace('\t', "    ");
+
+                            article.url = url.to_string();
+                            Ok(article)
+                        }
+                        Err(err) => {
+                            let stdout = std::str::from_utf8(&output.stdout)?;
+                            warn!("failed to deserialize {} into an `Article` struct:", stdout);
+                            Err(anyhow::anyhow!(err))
+                        }
+                    }
+                } else {
+                    let stderr = std::str::from_utf8(&output.stderr)?.to_string();
+                    Err(anyhow::anyhow!(stderr))
                 }
             }
-        } else {
-            let stderr = std::str::from_utf8(&output.stderr)?.to_string();
-            Err(anyhow::anyhow!(stderr))
+            Err(_) => {
+                // fallback to the `readable-readability` crate if the command fails
+                let html = ureq::AgentBuilder::new()
+                    .timeout(std::time::Duration::from_secs(config::get_config().client_timeout))
+                    .build()
+                    .get(url)
```

Comment:
This is a function inside `HNClient` which has a ureq client already. Can you update the function signature to take `&self` and use `self.client` instead?

Reply:
Addressed at [28d89b5](https://github.com/aome510/hackernews-TUI/pull/94/commits/28d89b5a9f9394a8a5d0721f401363b04e8d6bfc)